travis/self_tests

```diff
@@ -1,77 +1,406 @@
+        expected_errors.pop('class-camelcase')
+        expected_errors.pop('manifest-deprecated-key')
+        expected_errors.pop('copy-wo-api-one')
+        expected_errors['manifest-version-format'] = 5
+        self.pylint_run(
+            is_pr=True, version="7.0",
+            expected_errors=84, errors_dict=expected_errors)
+
+        expected_errors = EXPECTED_ERRORS.copy()
+        self.pylint_run(
+            is_pr=False, version="7.0",
+            expected_errors=27, errors_dict=expected_errors)
+
+    @unittest.skipIf(os.environ.get('LINT_CHECK', 0) != '1', "Set LINT_CHECK")
+    def test_check_v8(self):
+        print "VERSION=8.0"
+        expected_errors = EXPECTED_ERRORS.copy()
+        expected_errors.update(EXPECTED_ERRORS_PR)
+        self.pylint_run(
+            is_pr=True, version="8.0",
+            expected_errors=91, errors_dict=expected_errors)
+        expected_errors = EXPECTED_ERRORS.copy()
+        self.pylint_run(
+            is_pr=False, version="8.0",
+            expected_errors=27, errors_dict=expected_errors)
+
+    @unittest.skipIf(os.environ.get('LINT_CHECK', 0) != '1', "Set LINT_CHECK")
+    def test_check_without_version(self):
+        """PYLINT_EXPECTED_ERRORS="27"
+           VERSION=""
+        """
+        print "VERSION=''"
+        expected_errors = EXPECTED_ERRORS.copy()
+        expected_errors.update(EXPECTED_ERRORS_PR)
+        self.pylint_run(
+            is_pr=True, version="",
+            expected_errors=91, errors_dict=expected_errors)
+        expected_errors = EXPECTED_ERRORS.copy()
+        self.pylint_run(
+            is_pr=False, version="",
+            expected_errors=27, errors_dict=expected_errors)
+
+    def version_validate(self, version):
+        if version == "" and self.git_work_dir:
+            repo_path = os.path.join(self.git_work_dir, '.git')
+            branch_name = GitRun(repo_path).get_branch_name()
+            version = (branch_name.replace('_', '-').split('-')[:1]
+                       if branch_name else False)
+            version = version[0] if version else None
+        if not version:
+            print travis_helpers.yellow(
+                'Undefined environment variable'
+                ' `VERSION`.\nSet `VERSION` for '
+                'compatibility with guidelines by version.')
+        return version
+
+    def pylint_run(self, is_pr, version, expected_errors, errors_dict):
+        odoo_version = self.version_validate(version)
+        extra_params_cmd = self.get_extra_params(odoo_version)
+        extra_info = "extra_params_cmd %s " % extra_params_cmd
+        print extra_info
+        conf = ["--rcfile=%s" % self.pylint_rcfile]
+        cmd = conf + extra_params_cmd + self.modules_cmd
+        real_errors = self.pylint(cmd)
+        count_errors = sum(real_errors.values())
+        count_info = "count_errors %s" % count_errors
+        print count_info
+        if is_pr:
+            modules_changed = get_modules_changed(
```

Comment:
I don't know, the issue request was: test all versions with and without `TRAVIS_PULL_REQUEST`, so a set the var `is_pr` instead of change the env var `TRAVIS_PULL_REQUEST`